### ISSUE TITLE
Presence of maxLength in enum property causes schema build failures

### ISSUE THREAD

**Issue by lpraneis** [oxidecomputer/typify#125]
It seems that the presence of the `maxLength` field in a string enum property causes the addition of extra `super::` keywords in the codegen file. Below is the failing schema, the compile error, and a schema with the `maxLength` field removed that successfully builds.

failing schema:
```json
{
  "$id": "https://example.com/arrays.schema.json",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "example title",
  "type": "object",
  "properties": {
    "enum_property": {
      "description": "name",
      "type": "string",
      "maxLength" : 32,
      "enum": ["first", "second"]
    }
  }
}
```

compile error:
```
error[E0433]: failed to resolve: there are too many leading `super` keywords
   |
26 |             super::ExampleTitleEnumPropertyInner("first".to_string()),
   |             ^^^^^ there are too many leading `super` keywords

error[E0433]: failed to resolve: there are too many leading `super` keywords
   |
27 |             super::ExampleTitleEnumPropertyInner("second".to_string()),
   |             ^^^^^ there are too many leading `super` keywords

```

Changes required to write schema
```diff
{
  "$id": "https://example.com/arrays.schema.json",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "example title",
  "type": "object",
  "properties": {
    "enum_property": {
      "description": "name",
      "type": "string",
-     "maxLength" : 32,
      "enum": ["first", "second"]
    }
  }
}
```

**Comment by ahl** [comment]
Interesting; thanks for filing. I'll take a look.

**Comment by justinmmott** [comment]
The output in `new_type_with_enum.out` from https://github.com/oxidecomputer/typify/pull/127 compiles and matches the JSON schema. 

I wasn't really sure why that super was there in the first place, but the rest of the tests passed so made me more okay with removing it.

**Comment by ahl** [comment]
#147 fixed this specific case; #137 is going to address the original problem
